h2o-r/h2o-package/R/infogram.R

```diff
@@ -0,0 +1,526 @@
+  parms$ignored_columns <- args$x_ignore
+  parms$response_column <- args$y
+
+  if (!missing(model_id))
+    parms$model_id <- model_id
+  if (!missing(validation_frame))
+    parms$validation_frame <- validation_frame
+  if (!missing(seed))
+    parms$seed <- seed
+  if (!missing(nfolds))
+    parms$nfolds <- nfolds
+  if (!missing(fold_assignment))
+    parms$fold_assignment <- fold_assignment
+  if (!missing(fold_column))
+    parms$fold_column <- fold_column
+  if (!missing(ignore_const_cols))
+    parms$ignore_const_cols <- ignore_const_cols
+  if (!missing(score_each_iteration))
+    parms$score_each_iteration <- score_each_iteration
+  if (!missing(offset_column))
+    parms$offset_column <- offset_column
+  if (!missing(weights_column))
+    parms$weights_column <- weights_column
+  if (!missing(standardize))
+    parms$standardize <- standardize
+  if (!missing(distribution))
+    parms$distribution <- distribution
+  if (!missing(plug_values))
+    parms$plug_values <- plug_values
+  if (!missing(max_iterations))
+    parms$max_iterations <- max_iterations
+  if (!missing(stopping_rounds))
+    parms$stopping_rounds <- stopping_rounds
+  if (!missing(stopping_metric))
+    parms$stopping_metric <- stopping_metric
+  if (!missing(stopping_tolerance))
+    parms$stopping_tolerance <- stopping_tolerance
+  if (!missing(balance_classes))
+    parms$balance_classes <- balance_classes
+  if (!missing(class_sampling_factors))
+    parms$class_sampling_factors <- class_sampling_factors
+  if (!missing(max_after_balance_size))
+    parms$max_after_balance_size <- max_after_balance_size
+  if (!missing(max_runtime_secs))
+    parms$max_runtime_secs <- max_runtime_secs
+  if (!missing(custom_metric_func))
+    parms$custom_metric_func <- custom_metric_func
+  if (!missing(auc_type))
+    parms$auc_type <- auc_type
+  if (!missing(infogram_algorithm))
+    parms$infogram_algorithm <- infogram_algorithm
+  if (!missing(protected_columns))
+    parms$protected_columns <- protected_columns
+  if (!missing(net_information_threshold))
+    parms$net_information_threshold <- net_information_threshold
+  if (!missing(total_information_threshold))
+    parms$total_information_threshold <- total_information_threshold
+  if (!missing(safety_index_threshold))
+    parms$safety_index_threshold <- safety_index_threshold
+  if (!missing(relevance_index_threshold))
+    parms$relevance_index_threshold <- relevance_index_threshold
+  if (!missing(data_fraction))
+    parms$data_fraction <- data_fraction
+  if (!missing(top_n_features))
+    parms$top_n_features <- top_n_features
+  if (!missing(compute_p_values))
+    parms$compute_p_values <- compute_p_values
+
+  if (!missing(infogram_algorithm_params))
+      parms$infogram_algorithm_params <- as.character(toJSON(infogram_algorithm_params, pretty = TRUE))
+  h2o.show_progress() # enable progress bar explicitly
```

Comment:
I might be wrong but if the user sets `h2o.no_progress()` then this will not only do something that the user didn't want but also it will influence the rest of the program.

In explain, I use the following to temporary hide the progress bar, if we really want to ignore users wishes I'd suggest to modify the following snippet and use it so we don't change the behavior for the rest of the program.
```r
with_no_h2o_progress <- function(expr) {
  show_progress <- environment(h2o.no_progress)$
    .pkg.env$
    PROGRESS_BAR
  if (length(show_progress) == 0L || show_progress) {
    on.exit(h2o.show_progress())
  } else {
    on.exit(h2o.no_progress())
  }
  h2o.no_progress()
  force(expr)
}
```

Reply:
I guess I am not sure why we have `h2o.show_progress()` here because it seems like other algorithms don't have it.  @wendycwong did you add this for a particular reason, or was it just copy/pasted from some other algo skeleton code?   I remember I reported some issues with the progress bar a while ago when the R API was first created (I think it wasn't working) so maybe this was added to address that?

Reply:
I talked to @wendycwong: I will remove it.